### DISCUSSION TITLE
How to Include jQuery in GrapesJS and Handle Form Submission Using AJAX?

### DISCUSSION THREAD

**Discussion by Olzzonne** [GrapesJS/grapesjs#6174] [Q&A]
**Main Question**
I'm working with GrapesJS and want to include jQuery to submit a form via an AJAX request. How can I integrate jQuery into GrapesJS?

**Sub Question**
How to create a stepper form/ create state to the components?

**Comment by fredrickreuben** [comment]
Use the script prop of the component definition, dynamically add jQuery, so you can check to see if the jQuery already exists or not then add it, as in `script: () => {
  //check if jQuery is already added or not if not add it
if (window.jQuery) {
      setJQueryLoaded(true);
    } else {
      const script = document.createElement('script');
      script.src = 'https://code.jquery.com/jquery-3.6.0.min.js';
      script.async = true;
      document.body.appendChild(script);
    }
  // `this` is bound to the component element
  var el = this
 //form submission and Ajax code goes here
$(el).on('submit', function(e) {
  e.preventDefault(); // Prevent default form submission

  const formData = $(this).serialize(); // Collect form data

  $.ajax({
    url: $(this).attr('action'), // Use the form's action attribute for the URL
    method: $(this).attr('method'), // Use the form's method attribute (GET or POST)
    data: formData,
    success: function(response) {
      // Handle success
      console.log('Form submitted successfully:', response);
    },
    error: function(error) {
      // Handle error
      console.error('Form submission error:', error);
    }
  });
});
};`
